packages/api/cms-api/src/access-log/access-log.interceptor.ts

```diff
@@ -0,0 +1,90 @@
+import { CallHandler, ExecutionContext, Inject, Injectable, Logger, NestInterceptor, Optional } from "@nestjs/common";
+import { GqlExecutionContext } from "@nestjs/graphql";
+import { CurrentUser } from "src/user-permissions/dto/current-user";
+
+import { FILTER_AUTHENTICATED_REQUEST } from "./access-log.constants";
+import { FilterAuthenticatedRequest } from "./access-log.module";
+
+const IGNORED_PATHS = ["/dam/images/:hash/:fileId", "/dam/files/:hash/:fileId", "/dam/images/preview/:fileId", "/dam/files/preview/:fileId"];
+
+@Injectable()
+export class AccessLogInterceptor implements NestInterceptor {
+    protected readonly logger = new Logger(AccessLogInterceptor.name);
+
+    constructor(@Optional() @Inject(FILTER_AUTHENTICATED_REQUEST) private readonly filterAuthenticatedRequest?: FilterAuthenticatedRequest) {}
+
+    intercept(context: ExecutionContext, next: CallHandler) {
+        const requestType = context.getType().toString();
+
+        const requestData: string[] = [];
+        let ignored = false;
+
+        if (requestType === "graphql") {
+            const graphqlExecutionContext = GqlExecutionContext.create(context);
+            const graphqlContext = graphqlExecutionContext.getContext();
+
+            if (
+                this.filterAuthenticatedRequest &&
+                this.filterAuthenticatedRequest({
+                    user: graphqlContext.req.user,
+                    authHeader: graphqlContext.req.headers["authorization"] || graphqlContext.req.headers["Authorization"],
+                })
+            ) {
+                ignored = true;
+            }
+
+            requestData.push(`ip: ${graphqlContext.req.ip}`);
+            this.pushUserToRequestData(graphqlContext.req.user, requestData);
+
+            const gqlArgs = graphqlExecutionContext.getArgs();
+            const gqlInfo = graphqlExecutionContext.getInfo();
+
+            if (gqlInfo.operation.operation === "mutation") {
+                delete gqlArgs["input"];
+                delete gqlArgs["data"];
+            }
+
+            requestData.push(
+                ...[
+                    `operationType: ${gqlInfo.parentType}`,
+                    `operationName: ${gqlInfo.operation.name.value}`,
+                    `resolver function: ${gqlInfo.fieldName}`,
+                    `args: ${JSON.stringify(gqlArgs)}`,
+                ],
+            );
+        } else {
+            const httpContext = context.switchToHttp();
+            const httpRequest = httpContext.getRequest();
+
+            if (
+                IGNORED_PATHS.some((ignoredPath) => httpRequest.route.path.includes(ignoredPath)) ||
+                (this.filterAuthenticatedRequest &&
+                    this.filterAuthenticatedRequest({
+                        user: httpRequest.user,
+                        authHeader: httpRequest.headers["authorization"] || httpRequest.headers["Authorization"],
```

Comment:
Is there a use case for this? The authorization header can contain anything, from jwt over opaque tokens to nothing depending on the current implementation. I think everything which could be necessary is part of the user-object above.